### PR TITLE
Update iOS TestApp for autolinking

### PR DESCRIPTION
**Pull request by dhei**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

TestApp iOS is now on RN 0.60.0 and uses autolinking.

NOTE: With RN 0.60.0 natively supports CocoaPods integration (https://facebook.github.io/react-native/blog/2019/07/03/version-60#cocoapods-by-default) with autolinking, the best practice is to install/restore CocoaPods dependencies via `pod install` rather than checking in entire `ios/Pods/` directory. In the light of that, we are getting rid of `ios/Pods/` directory and adding it to git ignore.

### CODE REVIEW

File: TestApp/update-npm-packages.sh [hunk 1]
```diff
@@ -24,5 +24,8 @@ npm install
 echo "Build shared framework..."
 (cd ../AppCenterReactNativeShared/ios && SRCROOT=`pwd` ./build-fat-framework.sh)
 
-echo "Running jetify"
+echo "Running jetify to resolve AndroidX compatibility issues..."
 npx jetify
+
+echo "Running pod install..."
+(cd ios && pod install)
```
Comment (dhei) [inline TestApp/update-npm-packages.sh]: Note the new development workflow is to run `update-npm-packages.sh` script from TestApp directory.

With autolinking, `TestApp/update-link-to-appcenter.sh` script is no longer needed, hence deleted in this PR.

File: TestApp/app/App.js [hunk 2]
```diff
@@ -3,7 +3,6 @@
 
 import { AppState, Alert, Platform, ToastAndroid } from 'react-native';
 import { createBottomTabNavigator, createAppContainer } from 'react-navigation';
-import Toast from 'react-native-simple-toast';
```
Comment (dhei) [inline TestApp/app/App.js]: `react-native-simple-toast` npm package is not compatible with iOS autolinking and I am not sure if there's any workaround.

We use `react-native-simple-toast` in TestApp to send us toast message to improve the usability of the app. Without toast, we will need to check the XCode console logs or Android Studio Logcat.

Let's remove it for now and create an item in the backlog to fix it later.

File: TestApp/ios/TestApp/AppDelegate.m [hunk 3]
```diff
@@ -2,25 +2,25 @@
...
-#import <AppCenterReactNativeAnalytics/AppCenterReactNativeAnalytics.h>
-#import <AppCenterReactNativeCrashes/AppCenterReactNativeCrashes.h>
-#import <AppCenterReactNativeShared/AppCenterReactNativeShared.h>
 
 #import <React/RCTBridge.h>
 #import <React/RCTBundleURLProvider.h>
 #import <React/RCTRootView.h>
+
+#import <AppCenterReactNativeShared/AppCenterReactNativeShared.h>
+#import <AppCenterReactNative.h>
```
Comment (annakocheshkova) [inline TestApp/ios/TestApp/AppDelegate.m]: #import "AppCenterReactNative.h" ?

File: TestApp/app/App.js [hunk 4]
```diff
@@ -108,16 +107,13 @@ Crashes.setListener({
 
   onBeforeSending() {
     console.log('Will send crash. onBeforeSending is invoked.');
-    Toast.show('Sending crashes...');
```
Comment (annakocheshkova) [inline TestApp/app/App.js]: Any substitution for this..?

Comment (annakocheshkova) [reply TestApp/app/App.js]: Logging at least

File: AppCenterReactNativeShared/Products/local.podspec [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  s.author           = { 'Microsoft' => 'appcentersdk@microsoft.com' }
+
+  system("SRCROOT=#{__dir__}/../ios #{__dir__}/../prepare-local-podspec.sh")
+
+  s.source = { :http => "file://#{__dir__}/AppCenter-SDK-ReactNative-iOS-Pod-#{s.version}.zip"}
+
+  s.platform          = :ios, '9.0'
+  s.requires_arc      = true
+
+  s.vendored_frameworks = 'AppCenterReactNativeShared/AppCenterReactNativeShared.framework'
```
Comment (annakocheshkova) [inline AppCenterReactNativeShared/Products/local.podspec]: Oh, let's align the `=` signs :wink:

File: TestApp/ios/TestApp/AppDelegate.m (hunk 3 above)
Comment (dhei) [reply TestApp/ios/TestApp/AppDelegate.m]: It's C/C++ convention. I believe the rule of thumb is to use angle brackets `<>` to refer to a header file owned by the library and quotation marks `""` to refer to a header file in the same app. That is what we did in the past and we should do the same here.

We verified `#import "AppCenterReactNative.h"` and `#import <AppCenterReactNative.h>` both work in this case.

File: AppCenterReactNativeShared/Products/local.podspec (hunk 5 above)
Comment (dhei) [reply AppCenterReactNativeShared/Products/local.podspec]: I don't believe aligning `=` signs is a convention in `podspec`, but I decided to do it anyway.

File: TestApp/app/App.js (hunk 4 above)
Comment (dhei) [reply TestApp/app/App.js]: Good point. Replaced `Toast.show` with console logs.

File: AppCenterReactNativeShared/Products/local.podspec [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+Pod::Spec.new do |s|
+  s.name              = 'AppCenterReactNativeShared'
+  s.version           = '2.1.0'
+  s.summary           = 'React Native plugin for Visual Studio App Center'
+  s.license           = { :type => 'MIT',  :file => 'AppCenterReactNativeShared/LICENSE' }
+  s.homepage          = 'https://github.com/microsoft/appcenter-sdk-react-native'
+  s.documentation_url = "https://docs.microsoft.com/en-us/appcenter/"
+  s.author            = { 'Microsoft' => 'appcentersdk@microsoft.com' }
+  system("SRCROOT=#{__dir__}/../ios #{__dir__}/../prepare-local-podspec.sh")
```
Comment (jaelim-ms) [inline AppCenterReactNativeShared/Products/local.podspec]: What is this one?

File: AppCenterReactNativeShared/Products/local.podspec [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# This podspec is intended to be used as local pod reference during development,
```
Comment (jaelim-ms) [inline AppCenterReactNativeShared/Products/local.podspec]: Why do we need a local pod repo? Just curiosity.

File: AppCenterReactNativeShared/prepare-local-podspec.sh [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Build AppCenterReactNativeShared.framework
+(cd $SRCROOT; $SRCROOT/build-fat-framework.sh)
+
+# Generate AppCenter-SDK-ReactNative-iOS-Pod-{version}.zip
+podspecFile=local.podspec $SRCROOT/../zip-framework.sh
```
Comment (jaelim-ms) [inline AppCenterReactNativeShared/prepare-local-podspec.sh]: Can you explain this line?

File: AppCenterReactNativeShared/zip-framework.sh [hunk 9]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 cd Products
-wrapperSdkString=$(grep s.version AppCenterReactNativeShared.podspec)
+podspecFile="${podspecFile:-AppCenterReactNativeShared.podspec}"
```
Comment (jaelim-ms) [inline AppCenterReactNativeShared/zip-framework.sh]: Can you explain this line?

File: TestApp/ios/Podfile.lock [hunk 10]
```diff
@@ -1,4 +1,23 @@
 PODS:
+  - appcenter (2.1.0):
```
Comment (jaelim-ms) [inline TestApp/ios/Podfile.lock]: I wonder why this file is in the repo.

File: AppCenterReactNativeShared/Products/local.podspec (hunk 7 above)
Comment (dhei) [reply AppCenterReactNativeShared/Products/local.podspec]: We need the `local.podspec` with local reference for local development while keeping `AppCenterReactNativeShared.podspec` with official reference for Prod usage.

Prior to RN 0.60, we referencing the file path in the `Podfile` of TestApp like this (https://github.com/microsoft/appcenter-sdk-react-native/blob/2.1.0/TestApp/ios/Podfile#L15):

```ruby
pod 'AppCenterReactNativeShared', :path => '../../AppCenterReactNativeShared/Products/'
```

But with RN 0.60 autolinking, it is not possible to do so. We figured out the easiest way is to add a separate podspec for local development,

```ruby
pod 'AppCenterReactNativeShared', :podspec => '../../AppCenterReactNativeShared/Products/local.podspec'
```

To sum up, in local development, when you run `.\update-npm-packages.sh` from `TestApp` directory, the script calls `pod install` (https://github.com/microsoft/appcenter-sdk-react-native/pull/634/files#diff-8d44ab4d824b97b6ba9ca3cfacc4760eR34) which internally build and install from `local.podspec`.

File: AppCenterReactNativeShared/Products/local.podspec (hunk 6 above)
Comment (dhei) [reply AppCenterReactNativeShared/Products/local.podspec]: In order for `pod install` to work, we need to generate `AppCenter-SDK-ReactNative-iOS-Pod-{version}.zip` and this `system` call is to execute `prepare-local-podspec.sh` shell script to do so.

File: TestApp/ios/Podfile.lock (hunk 10 above)
Comment (dhei) [reply TestApp/ios/Podfile.lock]: It's best practice to check in Podfile.lock to ensure consistency of the development environment (cocoapods version, dependency version, etc.). With RN 0.60 it's even more useful as RN.iOS natively uses cocoapods to manage native dependencies.

File: AppCenterReactNativeShared/zip-framework.sh (hunk 9 above)
Comment (dhei) [reply AppCenterReactNativeShared/zip-framework.sh]: Will add a comment.

`podspecFile` is parameterized but its default value is `podspecFile:-AppCenterReactNativeShared.podspec` when it's not set.

File: AppCenterReactNativeShared/prepare-local-podspec.sh (hunk 8 above)
Comment (dhei) [reply AppCenterReactNativeShared/prepare-local-podspec.sh]: Passing `local.podspec` to `zip-framework.sh` script as I explained in https://github.com/microsoft/appcenter-sdk-react-native/pull/634/files#r306022447.

File: CHANGELOG.md [hunk 11]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 2.1.1 (Active Development)
 
+### App Center
+
+* **[Feature]** Support for React Native 0.60.0.
+
 ### AppCenterAuth
```
Comment (guperrot) [inline CHANGELOG.md]: We should align casing for all sections of this version

File: CHANGELOG.md [hunk 12]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 2.1.1 (Active Development)
```
Comment (guperrot) [inline CHANGELOG.md]: 2.2.0

Comment (dhei) [reply CHANGELOG.md]:
```suggestion
## Version 2.2.0 (Active Development)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,10 @@
 # App Center SDK for React Native Change Log
 
-## Version 2.1.1 (Active Development)
+## Version 2.2.0 (Active Development)
+
+### AppCenter
+
+* **[Feature]** Support React Native 0.60.0.
 
 ### AppCenterAuth
```
Comment (MatkovIvan) [inline CHANGELOG.md]: @dhei 
https://github.com/microsoft/appcenter-sdk-apple/pull/1691/files
